src/core/smallvec_type.hpp

```diff
@@ -14,375 +14,42 @@
-	 * @param count Number of consecutive items to remove.
-	 */
-	inline void ErasePreservingOrder(T *item, uint count = 1)
-	{
-		if (count == 0) return;
-		assert(item >= this->Begin());
-		assert(item + count <= this->End());
-
-		this->items -= count;
-		ptrdiff_t to_move = this->End() - item;
-		if (to_move > 0) MemMoveT(item, item + count, to_move);
-	}
-
-	/**
-	 * Tests whether a item is present in the vector, and appends it to the end if not.
-	 * The '!=' operator of T is used for comparison.
-	 * @param item Item to test for
-	 * @return true iff the item is was already present
-	 */
-	inline bool Include(const T &item)
-	{
-		bool is_member = this->Contains(item);
-		if (!is_member) *this->Append() = item;
-		return is_member;
-	}
-
-	/**
-	 * Get the number of items in the list.
-	 *
-	 * @return The number of items in the list.
-	 */
-	inline uint Length() const
-	{
-		return this->items;
-	}
-
-	/**
-	 * Get the pointer to the first item (const)
-	 *
-	 * @return the pointer to the first item
-	 */
-	inline const T *Begin() const
-	{
-		return this->data;
-	}
-
-	/**
-	 * Get the pointer to the first item
-	 *
-	 * @return the pointer to the first item
-	 */
-	inline T *Begin()
-	{
-		return this->data;
-	}
-
-	/**
-	 * Get the pointer behind the last valid item (const)
-	 *
-	 * @return the pointer behind the last valid item
-	 */
-	inline const T *End() const
-	{
-		return &this->data[this->items];
-	}
-
-	/**
-	 * Get the pointer behind the last valid item
-	 *
-	 * @return the pointer behind the last valid item
-	 */
-	inline T *End()
-	{
-		return &this->data[this->items];
-	}
-
-	/**
-	 * Get the pointer to item "number" (const)
-	 *
-	 * @param index the position of the item
-	 * @return the pointer to the item
-	 */
-	inline const T *Get(uint index) const
-	{
-		/* Allow access to the 'first invalid' item */
-		assert(index <= this->items);
-		return &this->data[index];
-	}
-
-	/**
-	 * Get the pointer to item "number"
-	 *
-	 * @param index the position of the item
-	 * @return the pointer to the item
-	 */
-	inline T *Get(uint index)
-	{
-		/* Allow access to the 'first invalid' item */
-		assert(index <= this->items);
-		return &this->data[index];
-	}
-
-	/**
-	 * Get item "number" (const)
-	 *
-	 * @param index the position of the item
-	 * @return the item
-	 */
-	inline const T &operator[](uint index) const
-	{
-		assert(index < this->items);
-		return this->data[index];
-	}
-
-	/**
-	 * Get item "number"
-	 *
-	 * @param index the position of the item
-	 * @return the item
-	 */
-	inline T &operator[](uint index)
-	{
-		assert(index < this->items);
-		return this->data[index];
-	}
-};
+template <typename T>
+inline bool include(std::vector<T>& vec, const T &item)
+{
+	const bool is_member = std::find(vec.begin(), vec.end(), item) != vec.end();
+	if (!is_member) vec.emplace_back(item);
+	return is_member;
+}
 
+/**
+ * Helper function to grow a vector if it is not already contained
```

Comment:
What does this even mean?
Suggestion: Add several new default-constructed items and return a pointer to the first.

Reply:
Lost hunk in merge, re-wrote, missed that line. Was tired, 11/10 would fail again.